pvlib/modelchain.py

```diff
@@ -770,30 +856,37 @@ def infer_aoi_model(self):
         else:
             raise ValueError('could not infer AOI model from '
                              'system.module_parameters. Check that the '
-                             'system.module_parameters contain parameters for '
+                             'module_parameters for all Arrays in '
+                             'system.arrays contain parameters for '
                              'the physical, aoi, ashrae or martin_ruiz model; '
                              'explicitly set the model with the aoi_model '
                              'kwarg; or set aoi_model="no_loss".')
 
     def ashrae_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi, iam_model='ashrae')
+        self.results.aoi_modifier = self.system.get_iam(
+            self.results.aoi, iam_model='ashrae')
         return self
 
     def physical_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi, iam_model='physical')
+        self.results.aoi_modifier = self.system.get_iam(self.results.aoi,
+                                                        iam_model='physical')
         return self
 
     def sapm_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi, iam_model='sapm')
+        self.results.aoi_modifier = self.system.get_iam(self.results.aoi,
+                                                        iam_model='sapm')
         return self
 
     def martin_ruiz_aoi_loss(self):
-        self.aoi_modifier = self.system.get_iam(self.aoi,
+        self.results.aoi_modifier = self.system.get_iam(self.results.aoi,
                                                 iam_model='martin_ruiz')
```

Comment:
I think stickler is using `flake8 --diff` internally. Line 882 might not be checked since it is unchanged. I'm not completely sure of how the `--diff` option works though. Good catch.